students/scott_bromley/lesson07/html_render.py

```diff
@@ -0,0 +1,154 @@
+#!/usr/bin/env python3
+
+"""
+A class-based system for rendering html.
+"""
+
+
+# This is the framework for the base class
+class Element(object):
+    tag = "html"
+    indent = "  "
+
+    def __init__(self, contents=None, **kwargs):
+        if contents is None:
+            self.contents = []
+        else:
+            self.contents = [contents]
+        if kwargs is None:
+            self.attributes = {}
+        else:
+            self.attributes = kwargs
+
+    def append(self, content):
+        if hasattr(content, 'render'):
+            self.contents.append(content)
+        else:
+            self.contents.append(TextWrapper(str(content)))
+
+    def render(self, out_file, cur_ind=""):
+        out_file.write(cur_ind + self.indent)
+        out_file.write(self._open_tag())
+        out_file.write("\n")
+        for content in self.contents:
+            try:
+                content.render(out_file, cur_ind + self.indent)
+            except (TypeError, AttributeError):
+                out_file.write(cur_ind + self.indent + content)
```

Comment:
you don't need the exception handling, just call content.render() always.

Reply:
also, the render() call will call to TextWrapper which only accepts one param - this is why you see the second error.